xwiki-platform-core/xwiki-platform-oldcore/src/test/java/com/xpn/xwiki/doc/XWikiDocumentTest.java

```diff
@@ -177,6 +178,14 @@ protected void setUp() throws Exception
         when(this.xWiki.getStore()).thenReturn(xWikiStoreInterface);
         when(this.xWiki.getDocument(any(DocumentReference.class), any())).thenReturn(this.document);
         when(this.xWiki.getDocumentReference(any(XWikiRequest.class), any())).thenReturn(documentReference);
+        when(this.xWiki.getDocumentReference((EntityReference) any(), any())).then(new Answer<DocumentReference>()
```

Comment:
Not sure but you might be able to do the same without the case by passing a class to `any`.
```suggestion
        when(this.xWiki.getDocumentReference(any(EntityReference.class), any())).then(new Answer<DocumentReference>()
```
Also, a matter a taste but the anonymous `Answer` can be converted to a lambda.

Reply:
> Not sure but you might be able to do the same without the case by passing a class to any.

Right, probably an old reflex of when any did not take a class. I could have looked at the line just above that one...

> Also, a matter a taste but the anonymous Answer can be converted to a lambda.

Indeed. sonalint usually suggest me this kind of improvement, but strangely it did not match this one.